x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/aggregation/blockhash/CategorizeRawBlockHash.java

```diff
@@ -19,33 +18,48 @@
 import org.elasticsearch.compute.data.Page;
 import org.elasticsearch.core.Releasable;
 import org.elasticsearch.core.Releasables;
-import org.elasticsearch.index.analysis.CharFilterFactory;
-import org.elasticsearch.index.analysis.CustomAnalyzer;
-import org.elasticsearch.index.analysis.TokenFilterFactory;
-import org.elasticsearch.index.analysis.TokenizerFactory;
+import org.elasticsearch.index.IndexService;
+import org.elasticsearch.index.analysis.AnalysisRegistry;
+import org.elasticsearch.xpack.core.ml.job.config.CategorizationAnalyzerConfig;
 import org.elasticsearch.xpack.ml.aggs.categorization.TokenListCategorizer;
 import org.elasticsearch.xpack.ml.job.categorization.CategorizationAnalyzer;
 
+import java.io.IOException;
+import java.util.List;
+
 /**
  * BlockHash implementation for {@code Categorize} grouping function.
  * <p>
  *     This implementation expects rows, and can't deserialize intermediate states coming from other nodes.
  * </p>
  */
 public class CategorizeRawBlockHash extends AbstractCategorizeBlockHash {
+    private static final CategorizationAnalyzerConfig ANALYZER_CONFIG = CategorizationAnalyzerConfig.buildStandardCategorizationAnalyzer(
+        List.of()
+    );
+
     private final CategorizeEvaluator evaluator;
 
-    CategorizeRawBlockHash(int channel, BlockFactory blockFactory, boolean outputPartial) {
+    CategorizeRawBlockHash(int channel, BlockFactory blockFactory, boolean outputPartial, AnalysisRegistry analysisRegistry) {
         super(blockFactory, channel, outputPartial);
-        CategorizationAnalyzer analyzer = new CategorizationAnalyzer(
-            // TODO: should be the same analyzer as used in Production
-            new CustomAnalyzer(
-                TokenizerFactory.newFactory("whitespace", WhitespaceTokenizer::new),
-                new CharFilterFactory[0],
-                new TokenFilterFactory[0]
-            ),
-            true
-        );
+
+        CategorizationAnalyzer analyzer;
+        try {
+            analyzer = new CategorizationAnalyzer(
+                analysisRegistry.buildCustomAnalyzer(
+                    IndexService.IndexCreationContext.RELOAD_ANALYZERS,
+                    null,
+                    false,
+                    ANALYZER_CONFIG.getTokenizer(),
+                    ANALYZER_CONFIG.getCharFilters(),
+                    ANALYZER_CONFIG.getTokenFilters()
+                ),
+                true
+            );
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
Added the `categorizer.close()`. 

I don't think it's worth the time writing a unit test. Next thing I'm doing is refactoring this anyway (after your and my PR are merged), so that the Abstrat/Raw/Intermediate BlockHash are one class. Then I'll construct the analyzer first, since it's the only thing that can fail

Reply:
Anything using BigArrays/CircuitBreaker _may_ fail. So even if the order is inverted, we would have to close whatever was created first, or there would be a circuitbreaker "memory leak".

Just to consider in the next PR!

Reply:
that makes sense. I'll handle it in the next (refactor) pr